File: algorithms/1_easy/P/00762_prime_number_of_set_bits_in_binary_representation/solution.cpp
class Solution {
    inline bool is_prime(const unsigned int &num) {
        switch (num) {
            case 2:
            case 3:
            case 5:
            case 7:
            case 11:
            case 13:
            case 17:
            case 19:
            case 23:
            case 29:
            case 31:
                return true;
            default:
                return false;
        }
    }

public:
    int countPrimeSetBits(int left, int right) {
        int count{0};

        for (int num{left}; num <=right; ++num) {
            if (is_prime(__builtin_popcount(num))) {
                count += 1;
            }
        }

        return count;
    }
};

File: algorithms/1_easy/P/00762_prime_number_of_set_bits_in_binary_representation/solution.rs
impl Solution {
    #[inline]
    fn is_prime(num: u32) -> bool {
        match num {
            2 | 3 | 5 | 7 | 11 | 13 | 17 | 19 | 23 | 29 | 31 => true,
            _ => false
        }
    }

    pub fn count_prime_set_bits(left: i32, right: i32) -> i32 {
        let mut count = 0_i32;

        for num in left..=right {
            if Self::is_prime(num.count_ones()) {
                count += 1;
            }
        }

        count
    }
}
